src/App.tsx

```diff
@@ -15,48 +17,125 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  length = 'length',
+  alphabet = 'alphabet',
+  default = 0,
+}
+
+function getSortedGoods(
+  goods: string[],
+  sortParameter: SortType,
+  reverseParameter: boolean,
+): string[] {
+  const copyGoods = [...goods];
+
+  if (reverseParameter) {
+    copyGoods.reverse();
+  }
+
+  if (sortParameter) {
+    copyGoods.sort((good1, good2) => {
+      switch (sortParameter) {
+        case SortType.length:
+          if (reverseParameter) {
+            return good2.length - good1.length;
+          }
+
+          return good1.length - good2.length;
+
+        case SortType.alphabet:
+          if (reverseParameter) {
+            return good2.localeCompare(good1);
+          }
+
+          return good1.localeCompare(good2);
+
+        default:
+          return SortType.default;
+      }
+    });
+  }
+
+  return copyGoods;
+}
+
 export const App: React.FC = () => {
+  const [reverseParameter, setReverseParameter] = useState(false);
+  const [sortParameter, setSortParameter] = useState(SortType.default);
+
+  const visibleGoods
+    = getSortedGoods(goodsFromServer, sortParameter, reverseParameter);
+
+  const reset = () => {
+    setSortParameter(SortType.default);
+    setReverseParameter(false);
+  };
+
+  const reverse = () => {
+    if (reverseParameter) {
+      setReverseParameter(false);
+    }
+
+    if (!reverseParameter) {
+      setReverseParameter(true);
+    }
+  };
```

Comment:
you can simplify using `prev` value
```suggestion
  const reverse = () => {
    setIsReversed(prev => !prev)
  };
```